File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Java/bba5ab22979b8d8cc56d18e78eae7cac.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    static class Point {
        double x, y;
        Point(double x, double y) {
            this.x = x;
            this.y = y;
        }
        Point() {
            this(0, 0);
        }
        Point add(Point q) {
            return new Point(x + q.x, y + q.y);
        }
        Point subtract(Point q) {
            return new Point(x - q.x, y - q.y);
        }
        Point multiply(double t) {
            return new Point(x * t, y * t);
        }
        Point divide(double t) {
            return new Point(x / t, y / t);
        }
        double dot(Point q) {
            return x * q.x + y * q.y;
        }
        double cross(Point q) {
            return x * q.y - y * q.x;
        }
        int compare(Point q) {
            int t = Double.compare(x, q.x);
            if (t != 0) return t;
            return Double.compare(y, q.y);
        }
        boolean equals(Point q) {
            return compare(q) == 0;
        }
        boolean notEquals(Point q) {
            return compare(q) != 0;
        }
        boolean lessThan(Point q) {
            return compare(q) < 0;
        }
        @Override
        public String toString() {
            return "(" + x + ", " + y + ")";
        }
    }

    static class Polygon extends ArrayList<Point> {}

    static final double EPS = 1e-8;

    static int compare(double x, double y, double tol) {
        return (x <= y + tol) ? (x + tol < y) ? -1 : 0 : 1;
    }

    static int ccw(Point p, Point q, Point r) {
        return compare((p.subtract(r)).cross(q.subtract(r)), 0, EPS);
    }

    static double angle(Point p, Point q, Point r) {
        Point u = p.subtract(q), v = r.subtract(q);
        return Math.atan2(u.cross(v), u.dot(v));
    }

    static double dist2(Point p, Point q) {
        Point diff = p.subtract(q);
        return diff.dot(diff);
    }

    static Point rotateCCW90(Point p) {
        return new Point(-p.y, p.x);
    }

    static Point rotateCW90(Point p) {
        return new Point(p.y, -p.x);
    }

    static Point rotateCCW(Point p, double t) {
        return new Point(p.x * Math.cos(t) - p.y * Math.sin(t), p.x * Math.sin(t) + p.y * Math.cos(t));
    }

    static Point projectPointLine(Point a, Point b, Point c) {
        return a.add(b.subtract(a).multiply(c.subtract(a).dot(b.subtract(a)) / b.subtract(a).dot(b.subtract(a))));
    }

    static Point projectPointSegment(Point a, Point b, Point c) {
        double r = b.subtract(a).dot(b.subtract(a));
        if (Math.abs(r) < EPS) return a;
        r = c.subtract(a).dot(b.subtract(a)) / r;
        if (r < 0) return a;
        if (r > 1) return b;
        return a.add(b.subtract(a).multiply(r));
    }

    static boolean between(Point p, Point q, Point r) {
        return ccw(p, q, r) == 0 && compare((p.subtract(q)).cross(r.subtract(q)), 0, EPS) <= 0;
    }

    static double distancePointSegment(Point a, Point b, Point c) {
        return Math.sqrt(dist2(c, projectPointSegment(a, b, c)));
    }

    static double distancePointPlane(double x, double y, double z, double a, double b, double c, double d) {
        return Math.abs(a * x + b * y + c * z - d) / Math.sqrt(a * a + b * b + c * c);
    }

    static boolean linesParallel(Point a, Point b, Point c, Point d) {
        return Math.abs((b.subtract(a)).cross(c.subtract(d))) < EPS;
    }

    static boolean linesCollinear(Point a, Point b, Point c, Point d) {
        return linesParallel(a, b, c, d) && Math.abs((a.subtract(b)).cross(a.subtract(c))) < EPS && Math.abs((c.subtract(d)).cross(c.subtract(a))) < EPS;
    }

    static boolean segIntersect(Point p, Point q, Point r, Point s) {
        Point A = q.subtract(p), B = s.subtract(r), C = r.subtract(p), D = s.subtract(q);
        int a = compare(A.cross(C)) + 2 * compare(A.cross(D));
        int b = compare(B.cross(C)) + 2 * compare(B.cross(D));
        if (a == 3 || a == -3 || b == 3 || b == -3) return false;
        if (a != 0 || b != 0 || p.equals(r) || p.equals(s) || q.equals(r) || q.equals(s)) return true;
        int t = (p.lessThan(r) ? 1 : 0) + (p.lessThan(s) ? 1 : 0) + (q.lessThan(r) ? 1 : 0) + (q.lessThan(s) ? 1 : 0);
        return t != 0 && t != 4;
    }

    static Point lineIntersect(Point p, Point q, Point r, Point s) {
        Point a = q.subtract(p), b = s.subtract(r), c = new Point(p.cross(q), r.cross(s));
        return new Point(new Point(a.x, b.x).cross(c), new Point(a.y, b.y).cross(c)).divide(a.cross(b));
    }

    static boolean circle2PtsRad(Point p1, Point p2, double r, Point c) {
        double d2 = (p1.x - p2.x) * (p1.x - p2.x) + (p1.y - p2.y) * (p1.y - p2.y);
        double det = ((r * r) / d2) - 0.25;
        if (det < 0.0) return false;
        double h = Math.sqrt(det);
        c.x = (p1.x + p2.x) * 0.5 + (p1.y - p2.y) * h;
        c.y = (p1.y + p2.y) * 0.5 + (p2.x - p1.x) * h;
        return true;
    }

    static Point computeCircleCenter(Point a, Point b, Point c) {
        b = a.add(b).divide(2);
        c = a.add(c).divide(2);
        return lineIntersect(b, b.add(rotateCW90(a.subtract(b))), c, c.add(rotateCW90(a.subtract(c))));
    }

    static Point computeCircleIncenter(Point a, Point b, Point c) {
        Point d = b.add(rotateCCW(c.subtract(b), angle(c, b, a) / 2));
        Point e = c.add(rotateCCW(a.subtract(c), angle(a, c, b) / 2));
        return lineIntersect(b, d, c, e);
    }

    static List<Point> circleLineIntersection(Point a, Point b, Point c, double r) {
        List<Point> ret = new ArrayList<>();
        b = b.subtract(a);
        a = a.subtract(c);
        double A = b.dot(b);
        double B = a.dot(b);
        double C = a.dot(a) - r * r;
        double D = B * B - A * C;
        if (D < -EPS) return ret;
        ret.add(c.add(a.add(b.multiply(-B + Math.sqrt(D + EPS))).divide(A)));
        if (D > EPS) ret.add(c.add(a.add(b.multiply(-B - Math.sqrt(D))).divide(A)));
        return ret;
    }

    static List<Point> circleCircleIntersection(Point a, Point b, double r, double R) {
        List<Point> ret = new ArrayList<>();
        double d = Math.sqrt(dist2(a, b));
        if (d > r + R || d + Math.min(r, R) < Math.max(r, R)) return ret;
        double x = (d * d - R * R + r * r) / (2 * d);
        double y = Math.sqrt(r * r - x * x);
        Point v = b.subtract(a).divide(d);
        ret.add(a.add(v.multiply(x).add(rotateCCW90(v).multiply(y))));
        if (y > 0) ret.add(a.add(v.multiply(x).subtract(rotateCCW90(v).multiply(y))));
        return ret;
    }

    static List<Point> tangentSegments(Point c, double r, Point p) {
        List<Point> resp = new ArrayList<>();
        Point toCenter = c.subtract(p);
        if (compare(r - Math.sqrt(dist2(toCenter, new Point())), 0, EPS) <= 0) {
            resp.add(p);
            resp.add(p);
            return resp;
        }
        double angle = Math.asin(r / Math.sqrt(dist2(toCenter, new Point())));
        Point uni = toCenter.divide(Math.sqrt(dist2(toCenter, new Point())));
        double tam = Math.sqrt(dist2(toCenter, new Point()) - r * r);
        Point origi = uni.multiply(tam);
        Point p1 = p.add(rotateCCW(origi, angle));
        resp.add(p1);
        Point p2 = p.add(rotateCCW(origi, -angle));
        resp.add(p2);
        return resp;
    }

    static int inPoly(Point p, Polygon T) {
        double a = 0;
        int N = T.size();
        for (int i = 0; i < N; i++) {
            if (between(T.get(i), p, T.get((i + 1) % N))) return -1;
            a += angle(T.get(i), p, T.get((i + 1) % N));
        }
        return compare(a, 0, EPS) != 0 ? 1 : 0;
    }

    static double polyArea(Polygon T) {
        double s = 0;
        int n = T.size();
        for (int i = 0; i < n; i++) {
            s += T.get(i).cross(T.get((i + 1) % n));
        }
        return s / 2;
    }

    static Point computeCentroid(List<Point> p) {
        Point c = new Point();
        double scale = 6.0 * polyArea((Polygon) p);
        for (int i = 0; i < p.size(); i++) {
            int j = (i + 1) % p.size();
            c = c.add(p.get(i).add(p.get(j)).multiply(p.get(i).x * p.get(j).y - p.get(j).x * p.get(i).y));
        }
        return c.divide(scale);
    }

    static boolean radialLt(Point p, Point q) {
        Point P = p.subtract(Point.pivot), Q = q.subtract(Point.pivot);
        double R = P.cross(Q);
        if (compare(R, 0, EPS) != 0) return R > 0;
        return compare(P.dot(P), Q.dot(Q), EPS) < 0;
    }

    static Polygon convexHull(List<Point> T) {
        int j = 0, k, n = T.size();
        Polygon U = new Polygon();
        Point.pivot = Collections.min(T, (a, b) -> a.compare(b));
        Collections.sort(T, (a, b) -> radialLt(a, b) ? -1 : 1);
        for (k = n - 2; k >= 0 && ccw(T.get(0), T.get(n - 1), T.get(k)) == 0; k--);
        Collections.reverse(T.subList(k + 1, n));
        for (int i = 0; i < n; i++) {
            while (j > 1 && ccw(U.get(j - 1), U.get(j - 2), T.get(i)) >= 0) j--;
            U.add(T.get(i));
            j++;
        }
        U.subList(j, U.size()).clear();
        return U;
    }

    static Polygon polyIntersect(Polygon P, Polygon Q) {
        int m = Q.size(), n = P.size();
        int a = 0, b = 0, aa = 0, ba = 0, inflag = 0;
        Polygon R = new Polygon();
        while ((aa < n || ba < m) && aa < 2 * n && ba < 2 * m) {
            Point p1 = P.get(a), p2 = P.get((a + 1) % n), q1 = Q.get(b), q2 = Q.get((b + 1) % m);
            Point A = p2.subtract(p1), B = q2.subtract(q1);
            int cross = compare(A.cross(B), 0, EPS);
            int ha = ccw(p2, q2, p1), hb = ccw(q2, p2, q1);
            if (cross == 0 && ccw(p1, q1, p2) == 0 && compare(A.dot(B), 0, EPS) < 0) {
                if (between(p1, q1, p2)) R.add(q1);
                if (between(p1, q2, p2)) R.add(q2);
                if (between(q1, p1, q2)) R.add(p1);
                if (between(q1, p2, q2)) R.add(p2);
                if (R.size() < 2) return new Polygon();
                inflag = 1;
                break;
            } else if (cross != 0 && segIntersect(p1, p2, q1, q2)) {
                if (inflag == 0) aa = ba = 0;
                R.add(lineIntersect(p1, p2, q1, q2));
                inflag = hb > 0 ? 1 : -1;
            }
            if (cross == 0 && hb < 0 && ha < 0) return R;
            boolean t = cross == 0 && hb == 0 && ha == 0;
            if (t ? inflag == 1 : cross >= 0 ? ha <= 0 : hb > 0) {
                if (inflag == -1) R.add(q2);
                ba++;
                b++;
                b %= m;
            } else {
                if (inflag == 1) R.add(p2);
                aa++;
                a++;
                a %= n;
            }
        }
        if (inflag == 0) {
            if (inPoly(P.get(0), Q) != 0) return P;
            if (inPoly(Q.get(0), P) != 0) return Q;
        }
        R = new Polygon(new HashSet<>(R));
        if (R.size() > 1 && R.get(0).equals(R.get(R.size() - 1))) R.remove(R.size() - 1);
        return R;
    }

    static final double PI = Math.acos(-1);

    static int xp, yp, vp, x, y, v, r;

    static Point origin = new Point(0, 0);

    static boolean pode(Point loc, double time) {
        Point nave = new Point(x, y);
        if (compare(distancePointSegment(loc, nave, origin), r, EPS) >= 0) {
            return compare(time, Math.abs(nave.subtract(loc).length()) / v, EPS) >= 0;
        } else {
            double distN = Math.sqrt(dist2(nave, origin) - r * r);
            double distL = Math.sqrt(dist2(loc, origin) - r * r);
            double a = Math.abs(nave.subtract(loc).length());
            double b = Math.abs(nave.length());
            double c = Math.abs(loc.length());
            double bigcos = (b * b + c * c - a * a) / (2 * b * c);
            double angNOL = Math.acos(bigcos) - Math.abs(Math.atan(distN / r)) - Math.abs(Math.atan(distL / r));
            double menor = distN + distL + Math.abs(r * angNOL);
            return compare(time, menor / v, EPS) >= 0;
        }
    }

    public static void main(String[] args) throws java.lang.Exception {
        Scanner sc = new Scanner(System.in);
        xp = sc.nextInt();
        yp = sc.nextInt();
        vp = sc.nextInt();
        x = sc.nextInt();
        y = sc.nextInt();
        v = sc.nextInt();
        r = sc.nextInt();
        double R = Math.hypot(yp, xp);
        double initial = Math.atan2(yp, xp);
        double bot = 0;
        double timeTot = (2 * PI * R) / vp;
        double top = 100000000000000.0;
        while (compare(top, bot, EPS) == 1) {
            double mid = (top + bot) / 2;
            double angle = ((2 * PI * mid) / timeTot) + initial;
            Point loc = new Point(Math.cos(angle) * R, Math.sin(angle) * R);
            if (pode(loc, mid)) {
                top = mid;
            } else {
                bot = mid;
            }
        }
        System.out.printf("%.7f\n", bot);
    }
}

File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Go/bba5ab22979b8d8cc56d18e78eae7cac.go
package main

import (
	"fmt"
	"math"
)

type point struct {
	x, y float64
}

func (p point) add(q point) point {
	return point{p.x + q.x, p.y + q.y}
}

func (p point) sub(q point) point {
	return point{p.x - q.x, p.y - q.y}
}

func (p point) mul(t float64) point {
	return point{p.x * t, p.y * t}
}

func (p point) div(t float64) point {
	return point{p.x / t, p.y / t}
}

func (p point) dot(q point) float64 {
	return p.x*q.x + p.y*q.y
}

func (p point) cross(q point) float64 {
	return p.x*q.y - p.y*q.x
}

func cmp(x, y, tol float64) int {
	if x <= y+tol {
		if x+tol < y {
			return -1
		}
		return 0
	}
	return 1
}

type polygon []point

func ccw(p, q, r point) int {
	return cmp((p.sub(r)).cross(q.sub(r)), 0, 0)
}

func angle(p, q, r point) float64 {
	u := p.sub(q)
	v := r.sub(q)
	return math.Atan2(u.cross(v), u.dot(v))
}

func dist2(p, q point) float64 {
	return p.sub(q).dot(p.sub(q))
}

func rotateCCW90(p point) point {
	return point{-p.y, p.x}
}

func rotateCW90(p point) point {
	return point{p.y, -p.x}
}

func rotateCCW(p point, t float64) point {
	cosT := math.Cos(t)
	sinT := math.Sin(t)
	return point{p.x*cosT - p.y*sinT, p.x*sinT + p.y*cosT}
}

func projectPointLine(a, b, c point) point {
	return a.add(b.sub(a).mul(c.sub(a).dot(b.sub(a)) / b.sub(a).dot(b.sub(a))))
}

func projectPointSegment(a, b, c point) point {
	r := b.sub(a).dot(b.sub(a))
	if math.Abs(r) < 1e-8 {
		return a
	}
	r = c.sub(a).dot(b.sub(a)) / r
	if r < 0 {
		return a
	}
	if r > 1 {
		return b
	}
	return a.add(b.sub(a).mul(r))
}

func between(p, q, r point) bool {
	return ccw(p, q, r) == 0 && cmp((p.sub(q)).cross(r.sub(q)), 0, 0) <= 0
}

func distancePointSegment(a, b, c point) float64 {
	return math.Sqrt(dist2(c, projectPointSegment(a, b, c)))
}

func distancePointPlane(x, y, z, a, b, c, d float64) float64 {
	return math.Abs(a*x+b*y+c*z-d) / math.Sqrt(a*a+b*b+c*c)
}

func linesParallel(a, b, c, d point) bool {
	return math.Abs((b.sub(a)).cross(c.sub(d))) < 1e-8
}

func linesCollinear(a, b, c, d point) bool {
	return linesParallel(a, b, c, d) && math.Abs((a.sub(b)).cross(a.sub(c))) < 1e-8 && math.Abs((c.sub(d)).cross(c.sub(a))) < 1e-8
}

func segIntersect(p, q, r, s point) bool {
	A := q.sub(p)
	B := s.sub(r)
	C := r.sub(p)
	D := s.sub(q)
	a := cmp(A.cross(C)) + 2*cmp(A.cross(D))
	b := cmp(B.cross(C)) + 2*cmp(B.cross(D))
	if a == 3 || a == -3 || b == 3 || b == -3 {
		return false
	}
	if a != 0 || b != 0 || p == r || p == s || q == r || q == s {
		return true
	}
	t := cmp(p, r) + cmp(p, s) + cmp(q, r) + cmp(q, s)
	return t != 0 && t != 4
}

func lineIntersect(p, q, r, s point) point {
	a := q.sub(p)
	b := s.sub(r)
	c := point{a.cross(b), r.sub(p).cross(s.sub(r))}
	return point{point{a.x, b.x}.cross(c), point{a.y, b.y}.cross(c)}.div(a.cross(b))
}

func circle2PtsRad(p1, p2 point, r float64) (point, bool) {
	d2 := (p1.x-p2.x)*(p1.x-p2.x) + (p1.y-p2.y)*(p1.y-p2.y)
	det := (r*r)/d2 - 0.25
	if det < 0.0 {
		return point{}, false
	}
	h := math.Sqrt(det)
	c := point{(p1.x + p2.x) * 0.5 + (p1.y-p2.y)*h, (p1.y + p2.y) * 0.5 + (p2.x-p1.x)*h}
	return c, true
}

func computeCircleCenter(a, b, c point) point {
	b = a.add(b).div(2)
	c = a.add(c).div(2)
	return lineIntersect(b, b.add(rotateCW90(a.sub(b))), c, c.add(rotateCW90(a.sub(c))))
}

func computeCircleIncenter(a, b, c point) point {
	d := b.add(rotateCCW(c.sub(b), angle(c, b, a)/2))
	e := c.add(rotateCCW(a.sub(c), angle(a, c, b)/2))
	return lineIntersect(b, d, c, e)
}

func circleLineIntersection(a, b, c point, r float64) []point {
	ret := []point{}
	b = b.sub(a)
	a = a.sub(c)
	A := b.dot(b)
	B := a.dot(b)
	C := a.dot(a) - r*r
	D := B*B - A*C
	if D < -1e-8 {
		return ret
	}
	ret = append(ret, c.add(a).add(b.mul(-B+math.Sqrt(D+1e-8))/A))
	if D > 1e-8 {
		ret = append(ret, c.add(a).add(b.mul(-B-math.Sqrt(D))/A))
	}
	return ret
}

func circleCircleIntersection(a, b point, r, R float64) []point {
	ret := []point{}
	d := math.Sqrt(dist2(a, b))
	if d > r+R || d+math.Min(r, R) < math.Max(r, R) {
		return ret
	}
	x := (d*d - R*R + r*r) / (2 * d)
	y := math.Sqrt(r*r - x*x)
	v := b.sub(a).div(d)
	ret = append(ret, a.add(v.mul(x)).add(rotateCCW90(v).mul(y)))
	if y > 0 {
		ret = append(ret, a.add(v.mul(x)).add(rotateCW90(v).mul(y)))
	}
	return ret
}

func tangentSegments(c point, r float64, p point) []point {
	resp := []point{}
	toCenter := c.sub(p)
	if cmp(r-math.Sqrt(dist2(toCenter, point{0, 0})), 0, 0) <= 0 {
		resp = append(resp, p)
		resp = append(resp, p)
		return resp
	}
	angle := math.Asin(r / math.Sqrt(dist2(toCenter, point{0, 0})))
	uni := toCenter.div(math.Sqrt(dist2(toCenter, point{0, 0})))
	tam := math.Sqrt(dist2(toCenter, point{0, 0}) - r*r)
	origi := uni.mul(tam)
	p1 := p.add(rotateCCW(origi, angle))
	resp = append(resp, p1)
	p2 := p.add(rotateCCW(origi, -angle))
	resp = append(resp, p2)
	return resp
}

func inPoly(p point, T polygon) int {
	a := 0
	N := len(T)
	for i := 0; i < N; i++ {
		if between(T[i], p, T[(i+1)%N]) {
			return -1
		}
		a += cmp(angle(T[i], p, T[(i+1)%N]), 0, 0)
	}
	if cmp(float64(a), 0, 0) != 0 {
		return 1
	}
	return 0
}

func polyArea(T polygon) float64 {
	s := 0.0
	n := len(T)
	for i := 0; i < n; i++ {
		s += T[i].cross(T[(i+1)%n])
	}
	return s / 2
}

func computeCentroid(p []point) point {
	c := point{0, 0}
	scale := 6.0 * polyArea(p)
	for i := 0; i < len(p); i++ {
		j := (i + 1) % len(p)
		c = c.add(p[i].add(p[j]).mul(p[i].x*p[j].y - p[j].x*p[i].y))
	}
	return c.div(scale)
}

func radialLt(p, q point) bool {
	P := p.sub(point{0, 0})
	Q := q.sub(point{0, 0})
	R := P.cross(Q)
	if cmp(R, 0, 0) != 0 {
		return R > 0
	}
	return cmp(P.dot(P), Q.dot(Q), 0) < 0
}

func convexHull(T []point) polygon {
	j := 0
	k := 0
	n := len(T)
	U := make(polygon, n)
	pointPivot := minElement(T)
	for i := 0; i < n; i++ {
		if between(T[0], T[n-1], T[i]) {
			k = n - 2
			for k >= 0 && ccw(T[0], T[n-1], T[k]) == 0 {
				k--
			}
			reverse(k+1, T)
			break
		}
	}
	for i := 0; i < n; i++ {
		for j > 1 && ccw(U[j-1], U[j-2], T[i]) >= 0 {
			j--
		}
		U[j] = T[i]
		j++
	}
	U = U[:j]
	return U
}

func polyIntersect(P, Q polygon) polygon {
	m := len(Q)
	n := len(P)
	a := 0
	b := 0
	aa := 0
	ba := 0
	inflag := 0
	R := make(polygon, 0)
	for (aa < n || ba < m) && aa < 2*n && ba < 2*m {
		p1 := P[a]
		p2 := P[(a+1)%n]
		q1 := Q[b]
		q2 := Q[(b+1)%m]
		A := p2.sub(p1)
		B := q2.sub(q1)
		cross := cmp(A.cross(B), 0, 0)
		ha := ccw(p2, q2, p1)
		hb := ccw(q2, p2, q1)
		if cross == 0 && ccw(p1, q1, p2) == 0 && cmp(A.dot(B), 0, 0) < 0 {
			if between(p1, q1, p2) {
				R = append(R, q1)
			}
			if between(p1, q2, p2) {
				R = append(R, q2)
			}
			if between(q1, p1, q2) {
				R = append(R, p1)
			}
			if between(q1, p2, q2) {
				R = append(R, p2)
			}
			if len(R) < 2 {
				return polygon{}
			}
			inflag = 1
			break
		} else if cross != 0 && segIntersect(p1, p2, q1, q2) {
			if inflag == 0 {
				aa = 0
				ba = 0
			}
			R = append(R, lineIntersect(p1, p2, q1, q2))
			if hb > 0 {
				inflag = 1
			} else {
				inflag = -1
			}
		}
		if cross == 0 && hb < 0 && ha < 0 {
			return R
		}
		t := cross == 0 && hb == 0 && ha == 0
		if t && (inflag == 1 || cross >= 0 && ha <= 0 || cross < 0 && hb > 0) {
			if inflag == -1 {
				R = append(R, q2)
			}
			ba++
			b++
			b %= m
		} else {
			if inflag == 1 {
				R = append(R, p2)
			}
			aa++
			a++
			a %= n
		}
	}
	if inflag == 0 {
		if inPoly(P[0], Q) == 1 {
			return P
		}
		if inPoly(Q[0], P) == 1 {
			return Q
		}
	}
	R = unique(R)
	if len(R) > 1 && R[0] == R[len(R)-1] {
		R = R[:len(R)-1]
	}
	return R
}

func unique(p polygon) polygon {
	m := make(map[point]bool)
	res := make(polygon, 0)
	for _, val := range p {
		if _, ok := m[val]; !ok {
			m[val] = true
			res = append(res, val)
		}
	}
	return res
}

func minElement(p []point) point {
	minP := p[0]
	for _, val := range p {
		if val.y < minP.y || (val.y == minP.y && val.x < minP.x) {
			minP = val
		}
	}
	return minP
}

func reverse(k int, p []point) {
	for i, j := 0, k; i < j; i, j = i+1, j-1 {
		p[i], p[j] = p[j], p[i]
	}
}

func main() {
	var xp, yp, vp, x, y, v, r int
	fmt.Scan(&xp, &yp, &vp, &x, &y, &v, &r)
	R := math.Hypot(float64(yp), float64(xp))
	initial := math.Atan2(float64(yp), float64(xp))
	bot := 0.0
	timeTot := (2 * math.Pi * R) / float64(vp)
	top := 100000000000000.0
	for cmp(top, bot, 0) == 1 {
		mid := (top + bot) / 2
		angle := ((2 * math.Pi * mid) / timeTot) + initial
		loc := point{math.Cos(angle) * R, math.Sin(angle) * R}
		if pode(loc, mid, x, y, v, r) {
			top = mid
		} else {
			bot = mid
		}
	}
	fmt.Printf("%.7f\n", bot)
}

func pode(loc point, time float64, x, y, v, r int) bool {
	nave := point{float64(x), float64(y)}
	if cmp(distancePointSegment(loc, nave, point{0, 0}), float64(r), 0) >= 0 {
		return cmp(time, math.Abs(nave.sub(loc))/float64(v), 0) >= 0
	} else {
		distN := math.Sqrt(dist2(nave, point{0, 0}) - float64(r)*float64(r))
		distL := math.Sqrt(dist2(loc, point{0, 0}) - float64(r)*float64(r))
		a := math.Abs(nave.sub(loc))
		b := math.Abs(nave)
		c := math.Abs(loc)
		bigcos := (b*b + c*c - a*a) / (2 * b * c)
		angNOL := math.Acos(bigcos) - math.Abs(math.Atan(distN/float64(r))) - math.Abs(math.Atan(distL/float64(r)))
		menor := distN + distL + math.Abs(float64(r)*angNOL)
		return cmp(time, menor/float64(v), 0) >= 0
	}
}
